src/components/Graph.js

```diff
@@ -12,39 +12,58 @@ export class Graph extends React.Component {
     this.state = {
       link: null,
       node: null,
-      simulation: null
+      simulation: null,
+      options: [],
+      namespace: null,
+      data: null
     };
 
     this.generateGraph = this.generateGraph.bind(this);
-    this.graphUpdate = this.graphUpdate.bind(this);
     this.onDateTimeSelect = this.onDateTimeSelect.bind(this);
     this.ticked = this.ticked.bind(this);
+    this.handleNamespaceChange = this.handleNamespaceChange.bind(this);
     this.nodeCircleRadius = 10;
   }
 
   componentDidMount() {
     this.prepareSvg();
-    this.loadData(this.generateGraph);
+    this.loadData();
   }
 
   onDateTimeSelect() {
-    this.loadData(this.graphUpdate);
+    this.loadData();
   }
 
-  loadData(cb) {
+  handleNamespaceChange(e) {
+    const namespace = e ? e.value : null;
+    this.setState({
+      namespace: namespace
+    }, () =>{
+      this.generateGraph(this.state.data);
+    });
+  }
+
+  loadData() {
     axios.get(process.env.REACT_APP_BACKEND_HOST + '/v1/graph')
       .then((response) => {
-        cb(response);
+        const namespaces = [...new Set(response.data.nodes.map(node => node.properties ? node.properties.namespace : null))];
+        const options = namespaces.map(namespace => ({
```

Comment:
Please, rename the variable to something more specific, for instance `namespaces` or `namespaceOptions`.